xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-default/src/main/java/org/xwiki/refactoring/internal/DefaultRefactoringConfiguration.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.refactoring.internal;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.configuration.ConfigurationSource;
+import org.xwiki.refactoring.RefactoringConfiguration;
+
+/**
+ * Default implementation of {@link RefactoringConfiguration}.
+ *
+ * @version $Id$
+ * @since 12.8RC1
+ */
+@Component
+@Singleton
+public class DefaultRefactoringConfiguration implements RefactoringConfiguration
+{
+    @Inject
+    private ConfigurationSource configurationSource;
+
+    @Override
+    public boolean canSkipRecyclebin()
+    {
+        // check if the wiki is configured to allow the recyclebin to be skipped, and if the user has the right to do
+        // so (ie, is an advanced user).
+        return this.configurationSource.getProperty("refactoring.recyclebin.skip", Boolean.FALSE);
```

Comment:
Same comment for the property, why not make it consistent and positive?

Reply:
Actually I think the property is not well named. This is not about sending the doc to the recycle bin or not, it's about allowing the user to decide whether to send docs to the recycle bin or not right?

Reply:
`refactoring.recyclebin.skip` is currently replaced by the `canSkipRecycleBin` property of `XWiki.Refactoring.RefactoringConfigurationClass`.